src/auth/oauth2client.ts

```diff
@@ -1316,6 +1361,13 @@ export class OAuth2Client extends AuthClient {
     return new LoginTicket(envelope, payload);
   }
 
+  /**
+   * Returns a Promise that resolves with DownscopedAccessTokenResponse
+   * type if refreshHandler() is defined.
+   * If there is no refresh handler callback set up, Promise resolves void.
+   */
+  async refreshHandler(): Promise<DownscopedAccessTokenResponse | void> {}
```

Comment:
Also this should reject when no tokens are returned instead of returning null. I think rejecting is better as the `refreshHandler` can provide a reason for no tokens being returned.

Reply:
ok, I add a check  for all 3 cases after calling refreshHandler and but calling setCredentials

Reply:
I added a check for token if not rejected, but would it be redundant with the check for downscopedCreds.access_token not being null?